packages/effect/src/stream/index.ts

```diff
@@ -444,6 +444,10 @@ function filter_<R, E, A>(stream: Stream<R, E, A>, f: F.Predicate<A>): Stream<R,
   );
 }
 
+export function filter<A, B extends A>(
```

Comment:
I also added a definition to _filter_ allowing you to refine the type of output a stream produces. To see an example, it is used on line `919` of this file where a stream of `Wrapped ` values is refined to a stream of `WrappedStream ` values.

Reply:
This is nice, there was a variant in here:

https://github.com/Matechs-Garage/matechs-effect/blob/master/packages/effect/src/stream/index.ts#L458

But this overloading is better